src/ai/backend/storage/api/manager.py

```diff
@@ -776,14 +778,32 @@ class Params(TypedDict):
     ) as params:
         await log_manager_api_entry(log, "mkdir", params)
         ctx: RootContext = request.app["ctx"]
+        vfid = params["vfid"]
+        parents = params["parents"]
+        exist_ok = params["exist_ok"]
+        relpath = params["relpath"]
+        relpaths = relpath if isinstance(relpath, list) else [relpath]
+
         async with ctx.get_volume(params["volume"]) as volume:
-            with handle_fs_errors(volume, params["vfid"]):
-                await volume.mkdir(
-                    params["vfid"],
-                    params["relpath"],
-                    parents=params["parents"],
-                    exist_ok=params["exist_ok"],
-                )
+            with handle_fs_errors(volume, vfid):
+                mkdir_tasks = [
+                    volume.mkdir(vfid, rpath, parents=parents, exist_ok=exist_ok)
+                    for rpath in relpaths
+                ]
+                result_group = await asyncio.gather(*mkdir_tasks, return_exceptions=True)
+
+                # if result group is all false throw error.
+                if all([isinstance(res, FileExistsError) for res in result_group]):
+                    raise FileExistsError("None of directories created because they already exist")
+
+                for result_or_exception in result_group:
+                    if isinstance(result_or_exception, BaseException):
+                        log.error(
+                            "%s",
+                            repr(result_or_exception),
+                            exc_info=result_or_exception,
+                        )
+
```

Comment:
Let's refer the `handle_fs_errors()` function. It catches all ValueError and OSError and return the error message in response. How about return an appended error message in http response?